py/nbf.py

```diff
@@ -0,0 +1,241 @@
+from typing import Optional
+
+# host -> device
+# TODO: 4-byte versions omitted
+OPCODE_WRITE_8 = 0x03
+OPCODE_READ_8 = 0x13
+OPCODE_FENCE = 0xfe
+OPCODE_FINISH = 0xff
+
+# device -> host
+OPCODE_FINISH = 0x80
+OPCODE_PUTCH = 0x82
+
+# CSR addresses
+ADDRESS_CSR_FREEZE = 0x0000200002
```

Comment:
NOTE: This has changed in more recent versions of BP

Reply:
Opened #2 to address post PR